addons/l10n_nl/models/account_journal.py

```diff
@@ -6,6 +6,19 @@
 class AccountJournal(models.Model):
     _inherit = 'account.journal'
 
+    @api.model
+    def _fill_missing_values(self, vals):
+        super()._fill_missing_values(vals)
+
+        if vals.get('type') != 'purchase':
+            return
+
+        company = self.env['res.company'].browse(vals['company_id'])
```

Comment:
You can't be sure `company_id` is the the values

Reply:
If the company is not present in the values, can I use self.env.company instead ?

Reply:
Nevermind, I saw how to do it in the parent model.